pkg/allocation/converters/converter.go

```diff
@@ -50,10 +50,10 @@ func ConvertAllocationRequestToGSA(in *pb.AllocationRequest) *allocationv1.GameS
 		}
 	}
 
-	if in.GetMetaPatch() != nil {
+	if in.GetMetadata() != nil {
```

Comment:
you should accept either of MetaPatch or Metadata being set, until MetaPatch is removed.
If both are set, reject the call.